### ISSUE TITLE
Cookie Jar

### ISSUE THREAD

**Issue by MaximeHeckel** [gorilla/websocket#165]
Is it possible / Would it make sens, to have the option to add a cookie jar to the Dialer like it is the case for the http.Client object?
(
```
cookieJar, _ := cookiejar.New(nil)

client := &http.Client{
    Jar: cookieJar,
}
```
)

I currently have to create a global `cookies []*http.Cookie` variable and fill it after each dial, then use that array again to send the cookies for the next dial

**Comment by garyburd** [comment]
Yes, it makes sense to add a cookie jar to dialer. Most of the code can be copied directly from [net/http](https://github.com/golang/go/blob/be48aa3f3a16006ab31c424487af352ca374afed/src/net/http/client.go#L140-L156).

I don't understand the comment about needing a global `cookies []*http.Cookie`.  Why not use a cookie jar?

**Comment by MaximeHeckel** [comment]
Ok I started implementing that exact code, you can expect a PR pretty soon.
That global cookie array is my current workaround to keep track on the cookies I'm setting with the server. I'd prefer the cookie jar, it's cleaner

**Comment by garyburd** [comment]
Fixed by 56d95f294029281baca3f01a3bce54cebd6e854c
